server/src/main/java/org/elasticsearch/action/admin/cluster/settings/TransportClusterUpdateSettingsAction.java

```diff
@@ -159,7 +159,7 @@ protected ClusterUpdateSettingsResponse newResponse(boolean acknowledged) {
 
             @Override
             public void onAllNodesAcked() {
-                if (changed) {
+                if (reroute) {
                     reroute(true);
                 } else {
                     super.onAllNodesAcked();
```

Comment:
Hmm I think this completes the listener with `AcknowledgedResponse.of(true)` which is not a `ClusterUpdateSettingsResponse` so that'll fail badly.

Reply:
Ok false alarm, the anonymous `ClusterUpdateSettingsTask` overrides `newResponse(boolean)` to return a `ClusterUpdateSettingsResponse`.

Reply:
ohh sorry so it does